AirCasting/SDSync/MobileSessions/SDCardMobileSessionsSavingService.swift

```diff
@@ -207,6 +210,47 @@ class SDCardMobileSessionsSavingService: SDCardMobileSessionssSaver {
         }
     }
     
+    private func addMeasurementLocation(deviceID: String, measurements: SDCardMeasurementsRow) -> CLLocationCoordinate2D? {
+
+        let isMini = deviceID.lowercased().contains(AirBeamDeviceType.airBeamMini.rawName)
+        
+        if isMini {
+            guard let sessionUUID else {
+                Log.error("[SD Sync] Session doesn't have sessionUUID")
+                fatalError()
+            }
+            do {
+                let session = try context.existingSession(uuid: sessionUUID)
+                let filteredStreams = session.allStreams.filter { MeasurementStreamEntity in
+                    !MeasurementStreamEntity.allMeasurements!.isEmpty
```

Comment:
I'm always suspicious with "!" in code. Can we remove it somehow?

Reply:
done

Reply:
done